src/x11/x.c

```diff
@@ -1014,7 +1070,44 @@ void x_setup(void)
         x_win_setup();
         x_cairo_setup();
         x_shortcut_grab(&settings.history_ks);
+}
+
+void x_win_round_corners(void)
+{
+        XWindowAttributes win_attr;
+        XGetWindowAttributes(xctx.dpy, xctx.win, &win_attr);
+
+        int width = win_attr.width + win_attr.border_width;
+        int height = win_attr.height + win_attr.border_width;
+
+        Pixmap mask = XCreatePixmap(xctx.dpy, xctx.win, width, height, 1);
+        XGCValues xgcv;
+
+        GC shape_gc = XCreateGC(xctx.dpy, mask, 0, &xgcv);
+
+        int rad = settings.corner_radius;
+        int dia = 2 * rad;
```

Comment:
The diameter of the corner should be capped to the window width/height so that it doesn't cause rendering weirdness in case it's bigger than the window size.